src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Unix.cs

```diff
@@ -33,9 +33,15 @@ internal static unsafe int ReadAtOffset(SafeFileHandle handle, Span<byte> buffer
                 // The Windows implementation uses ReadFile, which ignores the offset if the handle
                 // isn't seekable.  We do the same manually with PRead vs Read, in order to enable
                 // the function to be used by FileStream for all the same situations.
-                int result = handle.CanSeek ?
-                    Interop.Sys.PRead(handle, bufPtr, buffer.Length, fileOffset) :
-                    Interop.Sys.Read(handle, bufPtr, buffer.Length);
+                int result = Interop.Sys.PRead(handle, bufPtr, buffer.Length, fileOffset);
+
+                // Fallback to read if pread fails.
+                if (result == -1)
+                {
+                    Debug.Assert(Interop.Sys.GetLastErrorInfo().Error == Interop.Error.ENXIO || !handle.CanSeek); // We want to catch other errors and add unit tests for them.
+                    result = Interop.Sys.Read(handle, bufPtr, buffer.Length);
```

Comment:
Could this lead to getting invalid data back? e.g. a read is issued for fileOffset != current file position, it fails for some transient reason, and then we issue a Read for a different position than was intended.

Seems like we need to special-case just the errors we know are valid in this situation, and only for those fall back to a Read.

Reply:
Done.